share/models/meta.py

```diff
@@ -47,6 +49,31 @@ def __str__(self):
         return self.url
 
 
+class Subject(models.Model):
+    name = models.TextField(unique=True)
+    lineages = JSONField()
```

Comment:
Since we decided subjects could have multiple parents, `django-mptt` is out the window and getting the full lineage(s) of a subject will be slow and annoying. Since the subjects won't change except by admin intervention, it seems better not to worry about referential integrity and compute all the lineages up front.

I could add a `parents` many-to-many, so we could query for all descendents of a subject or whatever, but I feel like the `lineages` JSONField should stay to avoid reconstructing them over and over.